plugins/techdocs/src/home/components/TechDocsIndexPage.tsx

```diff
@@ -16,10 +16,13 @@
 
 import React from 'react';
 import { useOutlet } from 'react-router';
-import { DefaultTechDocsHome } from './DefaultTechDocsHome';
+import {
+  DefaultTechDocsHome,
+  DefaultTechDocsHomeProps,
+} from './DefaultTechDocsHome';
 
-export const TechDocsIndexPage = () => {
+export const TechDocsIndexPage = (props: DefaultTechDocsHomeProps) => {
```

Comment:
This perhaps suggests that the props should have been named `TechDocsIndexPageProps`. Could have renamed and marked the old one as deprecated, if you wanted to.

Reply:
Yes I think you have a good point here, @pjungermann can we rename `DefaultTechDocsHomeProps` to `TechDocsIndexPageProps` and continue to export that + mark `DefaultTechDocsHomeProps` as deprecated as part of this PR. That way one can use `TechDocsIndexPageProps` for any custom homepage if wanted and it will be a bit more generic.

Reply:
@freben @emmaindal 
I followed the same as at CatalogPage

```
export function CatalogPage(props: DefaultCatalogPageProps) {
  const outlet = useOutlet();

  return outlet || <DefaultCatalogPage {...props} />;
}
```

but if you prefer I can change it.

Reply:
Will just echo that it'd be awesome to make the change suggested by Emma before we merge (including deprecation of the existing type/name); it'll make it clearer if anyone needs to type a variable that's provided to `<TechDocsIndexPage />` if the name of the type corresponds to the component.